core/src/com/unciv/ui/popups/options/MultiplayerTab.kt

```diff
@@ -166,7 +166,17 @@ private fun addMultiplayerServerOptions(
         val passwordStatusTable = Table().apply {
             add(
                 if (settings.multiplayer.passwords.containsKey(settings.multiplayer.server)) {
-                    "Your userId is password secured"
+                    val userId = settings.multiplayer.userId
+                    val password = settings.multiplayer.passwords[settings.multiplayer.server] ?: ""
+                    val authStatus = UncivGame.Current.onlineMultiplayer.multiplayerServer.fileStorage()
```

Comment:
You're executing an http call and are dependent on the result to finish building the UI. This is a recipe for Application Not Responding on Android, if the call takes over 500ms.
Instead, you should set the label to Pending, and in another thread (with `Concurrency.run {  }`) do the http call, and when you have the result, set the label text to the final result in the main thread (with `Concurrency.runOnGLThread {  }`)

Reply:
ok, i have resolved the other requested changes. however I am unsure how to do this one.

Reply:
Something like

```kotlin

            val initialLabelText = if (settings.multiplayer.passwords.containsKey(settings.multiplayer.server)) "Working..."
            else "Set a password to secure your userId"
            val label = initialLabelText.toLabel()
            Concurrency.run {
                val response = getResponse() // Do all your stuff here to get the response
                val newText = something // get the new text
                Concurrency.runOnGLThread { label.setText(newText) }
            }
```

Reply:
resolved with https://github.com/yairm210/Unciv/pull/13198/commits/37db49c32883f0f1347e78d471dd1e8409d6985e